src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp

```diff
@@ -256,7 +257,13 @@ StorageObjectStorageQueue::StorageObjectStorageQueue(
         (*queue_settings_)[ObjectStorageQueueSetting::cleanup_interval_max_ms],
         getContext()->getServerSettings()[ServerSetting::keeper_multiread_batch_size]);
 
-    task = getContext()->getSchedulePool().createTask("ObjectStorageQueueStreamingTask", [this] { threadFunc(); });
+    size_t task_count = (*queue_settings_)[ObjectStorageQueueSetting::parallel_inserts] ? (*queue_settings_)[ObjectStorageQueueSetting::processing_threads_num] : 1;
```

Comment:
It seems to me it is better to have a separate setting defining number of "insert threads". So `parallel_inserts` setting could actually be `UInt` (by default equal to 1) to define the number of inserting threads.